spread/client.go

```diff
@@ -281,26 +281,32 @@ func (c *Client) run(script string, dir string, env *Environment, mode outputMod
 		rebootKey = rerr.Key
 		output = append(output, '\n')
 
-		timedout := time.After(c.killTimeout)
-		err := c.Run(fmt.Sprintf("reboot &\nsleep %.0f", c.killTimeout.Seconds()), "", nil)
+		uptime, err := c.getUptime()
 		if err != nil {
-			err = c.Run("echo should-have-disconnected", "", nil)
+			return nil, fmt.Errorf("failed to get uptime before reboot, error: %v", err)
 		}
-		if err == nil {
-			select {
-			case <-timedout:
-				return nil, fmt.Errorf("kill-timeout reached on %s while waiting for reboot", c.job)
-			default:
-			}
-			return nil, fmt.Errorf("reboot request on %s failed", c.job)
-		}
-		if err := c.dialOnReboot(); err != nil {
+		c.Run(fmt.Sprintf("reboot"), "", nil)
+		if err := c.dialOnReboot(uptime); err != nil {
```

Comment:
Let's please have a `time.Sleep(3 * time.Second)` before the line above. Dialing right after running the reboot command will add more churn to the rebooting system for no benefit.